tests/test_extract_random_seq_subset.py

```diff
@@ -0,0 +1,188 @@
+"""Unit tests for extract_random_seq_subset.py"""
+
+import os
+import stat
+import filecmp
+import shutil
+from htsinfer.extract_random_seq_subset import extract
+
+path = os.path.dirname(__file__)
+path = os.path.join(path, "test_files")
+outpath = os.path.join(path, "outputs")
+
+
+class TestExtract:
+    """Test `extract()` function."""
+
+    # set up output directory
+    if not os.path.exists(outpath):
+        os.makedirs(outpath)
+
+    def test_no_input_output_file(self):
+        """Test no input and no output file provided"""
+        assert extract([], []) == "no_input_output_provided"
+
+    def test_no_input_file(self):
+        """Test no input file provided"""
+        file1 = os.path.join(outpath, "SRR11971558_1_subset.fastq")
+        assert extract([], [file1]) == "no_input_provided"
+
+    def test_no_output_file(self):
+        """Test no output file provided"""
+        file1 = os.path.join(path, "SRR11971558_1.fastq.gz")
+        assert extract([file1], []) == "no_output_provided"
+
+    def test_output_file_error(self):
+        """Test unwritable output file"""
+        file1 = os.path.join(path, "SRR11971558_1.fastq.gz")
+        outfile1 = os.path.join(outpath, "no_seq.fastq")
+        with open(outfile1, 'w') as fp:
```

Comment:
I think there are ways how you can mock this without actually having to write anything, but I'd have to look these up. Guess it's not so important for now